File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    private val MAP = mapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")

    fun convert(n: Int) = MAP.filterKeys { n % it == 0 }
        .values.joinToString("").ifEmpty { "$n" }
}

File: scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  val MAP = Map(3 -> "Pling", 5 -> "Plang", 7 -> "Plong")

  def convert(n: Int) = List(MAP.filterKeys(n % _ == 0).values.mkString)
    .map(s => if (s.nonEmpty) s else s"$n").mkString

}

